MetaMorpheus/Test/GuiFunctionsTest.cs

```diff
@@ -48,5 +31,72 @@ public static void TestGetUniprotFilename(string proteomeID, bool reviewed, bool
             string filename = DownloadUniProtDatabaseFunctions.GetUniprotFilename(proteomeID, reviewed, isoforms, xmlFormat, compressed);
             Assert.AreEqual(expectedResult, filename);
         }
+
+        [Test]
+        [Parallelizable(ParallelScope.All)]
+        [TestCase("UP000000280", true, true, true, true, "1.fasta.gz", 50)]
+        [TestCase("UP000000280", true, true, true, false, "2.fasta", 50)]
+        [TestCase("UP000000280", true, true, false, true, "3.fasta.gz", 40)]
+        [TestCase("UP000000280", true, false, true, true, "4.xml.gz", 50)]
+        [TestCase("UP000000280", false, true, true, true, "5.fasta.gz", 2)]
+        [TestCase("UP000000280", true, true, false, false, "6.fasta", 50)]
+        [TestCase("UP000000280", true, false, true, false, "7.xml", 50)]
+        [TestCase("UP000000280", false, true, true, false, "8.fasta", 2)]
+        [TestCase("UP000000280", true, false, false, false, "9.fasta", 50)]
+        [TestCase("UP000000280", false, false, true, false, "10.xml", 2)]
+        [TestCase("UP000000280", false, false, false, false, "11.fasta", 2)]
+        public static async Task UniprotHtmlQueryTest(string proteomeID, bool reviewed, bool isoforms, bool xmlFormat, bool compressed,
+           string testName, int listCount)
+        {
+            var proteomeURL = DownloadUniProtDatabaseFunctions.GetUniProtHtmlQueryString(proteomeID, reviewed,
+                isoforms, xmlFormat, compressed);
+
+            var filePath = Path.Combine(TestContext.CurrentContext.TestDirectory, $@"DatabaseTests\{testName}");
+
+            List<Protein> reader = new List<Protein>();
+
+
+            try
```

Comment:
I don't think you need a try catch block - any unhandled assumption that's thrown will fail the test